### ISSUE TITLE
Location Change Not Functional Anymore

### ISSUE THREAD

**Issue by halalfood** [charliewolf/pynder#66]
When location is changed using this update_location, its supposed to change the location but it's not actually updating on tinder.

Reproducible steps:
1. Have a random location
2. Change location using method below
3. login to tinder on mobile app to verify change in settings

expected: new location in settings correlating to location above.
whats happening: location never changed

import pynder
import config

session = pynder.Session(config.FACEBOOK_ID, config.FACEBOOK_AUTH_TOKEN)
print 'connection made'

session.update_location(37.774, -122.419)

output:
connection made

Process finished with exit code 0

**Comment by SHABINTHEGREAT** [comment]
This is just an observation after having used the `session.update_location(LAT, LONG)` :
The location at the backend does seem to change, because further pynder calls bring in matches from the new location as updated; but the city name (place name) in the settings does not seem to reflect this.

Try with far locations' coordinates (London – NYC – LA). Check if the new matches are from the new location. Then also see the distance with the previous matches from the old locations, and check whether they correspond to the new distance.

**Comment by SHABINTHEGREAT** [comment]
**ISSUE CLARIFICATION**
I tried and looked at your issue. And here is the explanation.

There are three location related things on Tinder:
**`profile['pos']`**
**`profile['location']`**
**`profile['pos_info'] `**

**`update_location(LAT, LONG)`** — changes the system coordinates on the server side, which is the locaton used for finding matches.
This is given by — **`profile['pos']`**
The location name is given by — **`profile['location']`** — and seems to be updated only by the app, and has **nothing** to do with matching or the manually entered location using **`update_location(LAT, LONG)`**

The rest of **pynder works as promised**!

Extra bit of Info: There is also the **`profile['pos_info'] `** which gives further information alongwith **`profile['location'] `**— like the SE & NW extents to which the user's current city, state and country extends as Lat-Long coordinates.

_This issues may be Closed if no further clarification is needed._

**Comment by halalfood** [comment]
thanks - so I guess we need a way to change the location in the settings as well... Might be another project haha.

**Comment by halalfood** [comment]
Well I did some more testing. Updated location to the other side of the world. When it looks for people nearby, it is finding nothing - throwing a KeyError

**Comment by SHABINTHEGREAT** [comment]
Are you sure that is **_working_** Python? (I am not an expert, but that does not seem correct.)
Try the following and post the results, please:

1. `session.update_location( x[0][0], x[0][1] )`
2. `session.update_location( x[1][0], x[1][1] )`

**Comment by SHABINTHEGREAT** [comment]
Not a Pynder problem, at all.
Issue may be Closed!

**Comment by charliewolf** [comment]
Perhaps we need to update both `location` AND `pos`. Perhaps using a geocoder of some sort.

**Comment by halalfood** [comment]
y is consistent with numbers outside of the function so it can be 0,0 or 1,1 --- the point is this is NOT changing the position.
If you update_location - and then look for matches - the new matches are NOT in the new location.

**Comment by halalfood** [comment]
i agree with @charliewolf - there has to be something else that needs to be updated in order to change location.

**Comment by modulebaan** [comment]
I found a solution, write the coordinates as a string and it works:

session.update_location("37.774", "-122.419")

**Comment by modulebaan** [comment]
Sometimes it updates the location and sometimes it doesn't. 
I saw that if you wait enough it updates without any problems. You can't update your location every 5 mins it should be 10 or more.

**Comment by halalfood** [comment]
@modulebaan how long are we talking? an hour? hours? days?

**Comment by modulebaan** [comment]
Most of the time 15 minutes works but not always

**Comment by pointtonull** [comment]
I've found it wasn't working with me, but when I disabled my passport feature it started working again.

Try setting your passport option to "My current location".

**Comment by modulebaan** [comment]
I don't have Tinder Passport so it's always on 'My Current Location'
